lib/spack/spack/test/cmd/env.py

```diff
@@ -1921,6 +1921,39 @@ def test_env_include_concrete_remove_env():
     assert test2.path not in lockfile_as_dict["include_concrete"].keys()
 
 
+def test_env_include_concrete_reuse():
```

Comment:
I think we need tests for

1. reuse from all environments, but only from environments
2. reuse True includes from environments
3. reuse from one environment but not from another
4. test that reuse from an included environment that has been changed uses the specs from when the including environment was concretized.